### ISSUE TITLE
WASM support

### ISSUE THREAD

**Issue by pepealonso95** [xvrh/chrome_extension.dart#28]
Package needs to change dependencies like `'dart:js_util'` to `dart:js_interop` in order to be compatible with Flutter Web WASM.
https://dart.dev/interop/js-interop

**Comment by xvrh** [comment]
Yeah, the intent was to make it work with WASM and most of the generated bindings already uses `dart:js_interop`.

In practice, wasm compilation was never really tested and it's totally possible that I missed some requirements.

I was not aware that we couldn't use `dart:js_util`. I guess we will have to replace all the `promiseToFuture` with `toDart`.

**Comment by pepealonso95** [comment]
Any progress on including wasm support?

**Comment by xvrh** [comment]
No progress on my side. This task is to be taken up by anyone interested.

**Comment by EricPHassey** [comment]
Looks like web-renderer=html has been cut off in the beta build. I had to switch to the stable build to get it back. So just a matter of time we get left out, unless I'm missing something?

**Comment by xvrh** [comment]
@EricPHassey A normal `canvaskit` build works already. This issue is about supporting the `flutter build web --wasm` flag.

**Comment by bymoye** [comment]
> No progress on my side. This task is to be taken up by anyone interested.

It seems that the most used library in the project is `promiseToFuture<T>`, and `.toDart` should be used in `js_interop`

**Comment by pepealonso95** [comment]
@xvrh  I created a PR to be able to support --wasm. https://github.com/xvrh/chrome_extension.dart/pull/38 

Thanks for creating and maintaining this package :)
